src/physlr-split-minimizers.cc

```diff
@@ -0,0 +1,424 @@
+					}
+				}
+			}
+		}
+#if _OPENMP
+#pragma omp critical
+#endif
+		{
+			std::cout << ssOut.str();
+			std::cerr << ssErr.str();
+		}
+	}
+}
+
+void
+getBarcodeToMinimizer(
+    tsl::robin_map<std::string, BarcodeID>& barcodes,
+    std::vector<std::vector<Minimizer>>& barcodeToMinimizer,
+    std::string& inputFile)
+{
+	// vector of barcodes
+	std::vector<std::string> barcodeToStr;
+	std::string barcodeBuffer;
+	Minimizer minimizerBuffer;
+
+	// read in minimizer file
+	// format: GAGGTCCGTGGAGAGG-1	472493953667297251 1168973555595507959 342455687043295195
+	// 283275954102976652
+
+	std::ifstream fh;
+	fh.open(inputFile);
+	std::string line;
+	std::cerr << "Loading file " << inputFile << std::endl;
+	while (getline(fh, line)) {
+		std::stringstream ss(line);
+		ss >> barcodeBuffer;
+		const auto& barcode = barcodes.find(barcodeBuffer);
+		if (barcode == barcodes.end()) {
+			barcodeToStr.emplace_back(barcodeBuffer);
+			barcodes[barcodeBuffer] = barcodeToStr.size() - 1;
+			barcodeToMinimizer.emplace_back(std::vector<Minimizer>());
+			while (ss >> minimizerBuffer) {
+				barcodeToMinimizer[barcodeToStr.size() - 1].emplace_back(minimizerBuffer);
+			}
+		} else {
+			while (ss >> minimizerBuffer) {
+				barcodeToMinimizer[barcode->second].emplace_back(minimizerBuffer);
+			}
+		}
+	}
+}
+
+int
+main(int argc, char* argv[])
+{
+
+	bool die = false;
+	bool verbose = false;
+	static int help = 0;
+	static int version = 0;
+	int optindex = 0;
+
+	// long form arguments
+	static struct option longopts[] = { { "help", no_argument, &help, 1 },
+		                                { "threads", required_argument, nullptr, 't' },
+		                                { "version", no_argument, &version, 1 },
+		                                { nullptr, 0, nullptr, 0 } };
+
+	for (int c; (c = getopt_long(argc, argv, "t:", longopts, &optindex)) != -1;) {
+		switch (c) {
+		case 't': {
+			std::stringstream convert(optarg);
+			if (!(convert >> opt::threads)) {
+				std::cerr << "Error - Invalid parameters! t: " << optarg << std::endl;
+				return 0;
+			}
+			break;
+		}
+		case 'v': {
+			verbose = true;
+			break;
+		}
+		default: {
+			die = true;
+			break;
+		}
+		}
+	}
+	if (help != 0) {
+		printVersion();
+		exit(EXIT_SUCCESS);
+	} else if (version != 0) {
+		printHelpDialog();
+		exit(EXIT_SUCCESS);
+	}
+
+#if _OPENMP
+	omp_set_num_threads(opt::threads);
+#endif
+
+	// Stores fasta input file names
+	std::vector<std::string> inputFiles;
+
+	while (optind < argc) {
+		inputFiles.emplace_back(argv[optind]);
+		optind++;
+	}
+
+	if (inputFiles.empty()) {
+		printErrorMsg(PROGRAM, "missing file operand");
+		die = true;
+	}
+
+	if (die) {
+		printHelpDialog();
+		exit(EXIT_FAILURE);
+	}
+
+	// constuct minimizers to barcodes
+	// barcode to ID table (index in vector)
+	tsl::robin_map<std::string, BarcodeID> barcodes;
+
+	// barcodeID (index) to minimizer vector of vector
+	// Note: Because a vector isn't a set, the input cannot have duplicates.
```

Comment:
It is possible that the input has duplicate minimizers, right? Would that be an issue if that happened (other than the increased memory)

Reply:
How so. The results should be filtered and merged at that point so there would be no duplicate minimizers per barcode. This is the same assumption that `physlr-overlap` makes.

Reply:
Ah! Yup you're right -- I was thinking about the indexlr file with 1 read per line.